osu.Game/Overlays/Settings/Sections/Audio/AudioOffsetAdjustControl.cs

```diff
@@ -148,17 +149,28 @@ private void updateDisplay(object? _, NotifyCollectionChangedEventArgs e)
                         break;
                 }
 
-                SuggestedOffset.Value = averageHitErrorHistory.Any() ? averageHitErrorHistory.Average(dataPoint => dataPoint.SuggestedGlobalAudioOffset) : null;
+                SuggestedOffset.Value = averageHitErrorHistory.Any() ? Math.Round(averageHitErrorHistory.Average(dataPoint => dataPoint.SuggestedGlobalAudioOffset)) : null;
             }
 
             private float getXPositionForOffset(double offset) => (float)(Math.Clamp(offset, current.MinValue, current.MaxValue) / (2 * current.MaxValue));
 
             private void updateHintText()
             {
-                hintText.Text = SuggestedOffset.Value == null
-                    ? AudioSettingsStrings.SuggestedOffsetNote
-                    : AudioSettingsStrings.SuggestedOffsetValueReceived(averageHitErrorHistory.Count, SuggestedOffset.Value.ToLocalisableString(@"N0"));
-                applySuggestion.Enabled.Value = SuggestedOffset.Value != null;
+                if (SuggestedOffset.Value == null)
+                {
+                    applySuggestion.Enabled.Value = false;
+                    hintText.Text = AudioSettingsStrings.SuggestedOffsetNote;
+                }
+                else if (Math.Abs(SuggestedOffset.Value.Value - current.Value) < 1)
+                {
+                    applySuggestion.Enabled.Value = false;
+                    hintText.Text = AudioSettingsStrings.SuggestedOffsetCorrect(averageHitErrorHistory.Count);
+                }
+                else
+                {
+                    applySuggestion.Enabled.Value = true;
+                    hintText.Text = AudioSettingsStrings.SuggestedOffsetValueReceived(averageHitErrorHistory.Count, SuggestedOffset.Value.ToLocalisableString(@"N0"));
```

Comment:
One of the test scenes added here demonstrates that it's possible for this text to print "-0"ms:

![Screenshot 2025-04-17 at 11 26 09](https://github.com/user-attachments/assets/240d7ac4-209e-4065-8632-83fa2db7bd0f)

[Previously.](https://github.com/ppy/osu/pull/23367) Thank the IEEE semantics yet again (not).

I'm not sure there's an elegant fix for this, but I figure it's probably worth bringing this up before someone else does?

Reply:
No doubt, but I'd rather just get in front of it...